core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/EmptyPatternOptimizer.java

```diff
@@ -58,7 +71,10 @@ public void meet(StatementPattern node)
 		Value obj = node.getObjectVar().getValue();
 		Resource[] ctx = getContexts(node.getContextVar());
 		for (RepositoryConnection member : members) {
-			if (member.hasStatement(subj, pred, obj, true, ctx)) {
+			RepositoryBloomFilter bloomFilter = MoreObjects.firstNonNull(
```

Comment:
...though I can see that in other places this may make the code a bit more complex. So not too hung up on changing this - if you prefer the current approach just leave it.